### ISSUE TITLE
Calculation methods call completion handlers on inconsistent queues

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-directions-swift#411]
As of #382, sometimes `calculate(_:completionHandler:)` and `calculateRoutes(matching:completionHandler:)` call the completion handler on the URLSession delegate queue:

https://github.com/mapbox/mapbox-directions-swift/blob/be90d21c93c3047ea8f04b5f53d99f3d3e0e159b/Sources/MapboxDirections/Directions.swift#L160-L173

whereas sometimes these methods call the completion handler on the main queue:

https://github.com/mapbox/mapbox-directions-swift/blob/be90d21c93c3047ea8f04b5f53d99f3d3e0e159b/Sources/MapboxDirections/Directions.swift#L181-L183 https://github.com/mapbox/mapbox-directions-swift/blob/be90d21c93c3047ea8f04b5f53d99f3d3e0e159b/Sources/MapboxDirections/Directions.swift#L196-L198 https://github.com/mapbox/mapbox-directions-swift/blob/be90d21c93c3047ea8f04b5f53d99f3d3e0e159b/Sources/MapboxDirections/Directions.swift#L200-L203

The methods should consistently call the completion handler from a single queue, and we should document which queue that is. Previously, in v0.30.0, these methods always called the completion handler on the main queue:

https://github.com/mapbox/mapbox-directions-swift/blob/65857d14080afa1e5463e0dbadd9dec23e66a6f3/Sources/MapboxDirections/MBDirections.swift#L257-L259 https://github.com/mapbox/mapbox-directions-swift/blob/65857d14080afa1e5463e0dbadd9dec23e66a6f3/Sources/MapboxDirections/MBDirections.swift#L263-L265

As it is, a developer doesn’t know whether their completion handler implementation needs to dispatch to the main queue or stay on the current queue. So, for example, a completion handler that presents a UIAlertController would result in a crash like this:

```
#0	(null) in __exceptionPreprocess ()
#1	(null) in objc_exception_throw ()
#2	(null) in _AssertAutolayoutOnAllowedThreadsOnly ()
#3	(null) in -[NSISEngine withBehaviors:performModifications:] ()
#4	(null) in -[UIView(UIConstraintBasedLayout) _calculatedSystemLayoutSizeFittingSize:withHorizontalFittingPri... ()
#5	(null) in -[UIView(AdditionalLayoutSupport) _systemLayoutSizeFittingSize:withHorizontalFittingPriority:vert... ()
#6	(null) in -[_UIAlertControllerView _minimumSizeForAllActions] ()
#7	(null) in -[_UIAlertControllerView _itemSizeForHorizontalLayout:visualStyleRequiresActionRepresentationToFi... ()
#8	(null) in -[_UIAlertControllerView _canLayOutActionsHorizontally] ()
#9	(null) in -[_UIAlertControllerView _actionLayoutDirectionChanged] ()
#10	(null) in -[_UIAlertControllerView _updateStyleForIdiomChange:] ()
#11	(null) in -[_UIAlertControllerView _setVisualStyle:] ()
#12	(null) in -[UIAlertController loadView] ()
#13	(null) in -[UIViewController loadViewIfRequired] ()
#14	(null) in -[UIViewController view] ()
#15	(null) in -[UIViewController _setPresentationController:] ()
#16	(null) in -[UIViewController _presentViewController:modalSourceViewController:presentationController:animat... ()
#17	(null) in -[UIViewController _presentViewController:withAnimationController:completion:] ()
#18	(null) in __63-[UIViewController _presentViewController:animated:completion:]_block_invoke ()
#19	(null) in -[UIViewController _performCoordinatedPresentOrDismiss:animated:] ()
#20	(null) in -[UIViewController _presentViewController:animated:completion:] ()
#21	(null) in -[UIViewController presentViewController:animated:completion:] ()
```

/cc @mapbox/navigation-ios

**Comment by 1ec5** [comment]
Oh cool, I hadn’t noticed #410, which fixes this issue.

**Comment by 1ec5** [comment]
Fixed in #410.
